### PR TITLE
Move dcr and oauth scope API java code as separate module

### PR DESCRIPTION
**Pull request by senthalan**
### Proposed changes in this pull request

Move dcr and oauth scope API java code as a separate module.
Related to https://github.com/wso2/product-is/issues/11425

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.dcr/pom.xml [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>1.7</source>
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.api.server.dcr/pom.xml:152]: 1.8?

File: components/org.wso2.carbon.identity.oauth.scope.endpoint/pom.xml [hunk 2]
```diff
@@ -157,19 +152,14 @@
         </dependency>
         <dependency>
             <groupId>org.wso2.carbon.identity.inbound.auth.oauth2</groupId>
-            <artifactId>org.wso2.carbon.identity.oauth</artifactId>
+            <artifactId>org.wso2.carbon.identity.api.server.oauth.scope</artifactId>
             <exclusions>
                 <exclusion>
-                    <groupId>javax.ws.rs</groupId>
-                    <artifactId>jsr311-api</artifactId>
+                    <groupId>*</groupId>
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.scope.endpoint/pom.xml:158]: Why do we have * here?

Comment (senthalan) [reply components/org.wso2.carbon.identity.oauth.scope.endpoint/pom.xml:158]: We want to exclude all the transitive dependencies from the new module being packed inside the lib of the webapp.

File: components/org.wso2.carbon.identity.api.server.dcr/pom.xml (hunk 1 above)
Comment (senthalan) [reply components/org.wso2.carbon.identity.api.server.dcr/pom.xml]: Changed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1573/commits/78ec5de8160e7e362609783e01fafd4fca85afef

**Review by mefarazath — APPROVED** [review]

**Pull request merged**